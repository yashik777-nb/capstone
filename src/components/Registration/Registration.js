import React from "react";

export default class Registration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      firstName: "",
      lastName: "",
      location: "",
      mobileNumber: "",
    };
  }

  onSubmit(e) {
    e.preventDefault();
    const user = { ...this.state };
    this.props.onSave(user);
  }

  usernameChangeHandler(e) {
    this.setState({
      username: e.target.value,
    });
  }

  passwordChangeHandler(e) {
    this.setState({
      password: e.target.value,
    });
  }

  firstNameChangeHandler(e) {
    this.setState({
      firstName: e.target.value,
    });
  }

  lastNameChangeHandler(e) {
    this.setState({
      lastName: e.target.value,
    });
  }

  locationChangeHandler(e) {
    this.setState({
      location: e.target.value,
    });
  }

  mobileNumberChangeHandler(e) {
    this.setState({
      mobileNumber: e.target.value,
    });
  }

  render() {
    return (
      <main className="login-form m-5">
        <div className="container">
          <div className="row justify-content-center">
            <div className="col-md-8">
              <div className="card">
                <div className="card-header">
                  <p>Registration</p>
                </div>
                <div className="card-body">
                  <form onSubmit={(e) => this.onSubmit(e)}>
                    <div className="form-group row">
                      <label
                        htmlFor="email_address"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        E-Mail Address
                      </label>
                      <div className="col-md-6">
                        <input
                          type="text"
                          id="email_address"
                          className="form-control"
                          name="email-address"
                          onChange={(e) => this.usernameChangeHandler(e)}
                          required
                        />
                      </div>
                    </div>
                    <div className="form-group row">
                      <label
                        htmlFor="password"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        Password
                      </label>
                      <div className="col-md-6">
                        <input
                          type="password"
                          id="password"
                          className="form-control"
                          name="password"
                          onChange={(e) => this.passwordChangeHandler(e)}
                          required
                        />
                      </div>
                    </div>
                    <div className="form-group row">
                      <label
                        htmlFor="firstName"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        First Name
                      </label>
                      <div className="col-md-6">
                        <input
                          type="text"
                          id="firstName"
                          className="form-control"
                          name="firstName"
                          onChange={(e) => this.firstNameChangeHandler(e)}
                          required
                        />
                      </div>
                    </div>
                    <div className="form-group row">
                      <label
                        htmlFor="lastName"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        Last Name
                      </label>
                      <div className="col-md-6">
                        <input
                          type="text"
                          id="lastName"
                          className="form-control"
                          name="lastName"
                          onChange={(e) => this.lastNameChangeHandler(e)}
                          required
                        />
                      </div>
                    </div>
                    <div className="form-group row">
                      <label
                        htmlFor="location"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        Location
                      </label>
                      <div className="col-md-6">
                        <input
                          type="text"
                          id="location"
                          className="form-control"
                          name="location"
                          onChange={(e) => this.locationChangeHandler(e)}
                        />
                      </div>
                    </div>
                    <div className="form-group row">
                      <label
                        htmlFor="mobileNumber"
                        className="col-md-4 col-form-label text-md-right"
                      >
                        Mobile Number
                      </label>
                      <div className="col-md-6">
                        <input
                          type="number"
                          id="mobileNumber"
                          className="form-control"
                          name="mobileNumber"
                          onChange={(e) => this.mobileNumberChangeHandler(e)}
                        />
                      </div>
                    </div>
                    <div className="col-md-6 offset-md-4">
                      <button
                        type="submit"
                        className="btn btn-primary btn-block"
                      >
                        Register
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    );
  }
}
