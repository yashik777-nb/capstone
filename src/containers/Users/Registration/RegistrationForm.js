import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Registration from "../../../components/Registration/Registration";
import * as actionCreator from "../../../store/actions/actionsIndex";
class RegistrationForm extends React.Component {
  registerUserHandler(user) {
    this.props.registerUser(user);
    this.props.history.push("/");
  }
  render() {
    return <Registration onSave={(user) => this.registerUserHandler(user)} />;
  }
}

const mapDispathToProps = (dispatch) => {
  return {
    registerUser: (user) => dispatch(actionCreator.addUser(user)),
  };
};

export default connect(null, mapDispathToProps)(withRouter(RegistrationForm));
