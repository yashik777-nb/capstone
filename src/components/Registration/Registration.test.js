import React from "react";
import { configure } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import moxios from "moxios";
import toJson from "enzyme-to-json";

import Registration from "./Registration";

configure({ adapter: new Adapter() });

const styles = {
  paper: "",
  avatar: "",
  form: "",
  submit: "",
  register: "",
};

const props = {
  classes: styles,
};

const mockFn = jest.fn();

describe("Registration Component Tests", (done) => {
  let wrapper;

  beforeEach(() => {
    const mount = createMount();
    wrapper = mount(
      <BrowserRouter>
        <Registration {...props} onSave={mockFn} />
      </BrowserRouter>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("Should Have Username Required for Registartion", () => {
    expect(wrapper.find(TextField).at(0).props().required).toEqual(true);
  });

  it("Registration Form Test", () => {
    wrapper
      .find(TextField)
      .at(0)
      .simulate("change", {
        target: { value: "yash@abc.com" },
      });
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", {
        target: { value: "yash" },
      });
    wrapper
      .find(TextField)
      .at(2)
      .simulate("change", {
        target: { value: "Yash" },
      });
    wrapper
      .find(TextField)
      .at(3)
      .simulate("change", {
        target: { value: "IK" },
      });
    wrapper
      .find(TextField)
      .at(4)
      .simulate("change", {
        target: { value: "UK" },
      });
    wrapper
      .find(TextField)
      .at(1)
      .simulate("change", {
        target: { value: 12121212 },
      });
    wrapper.update();
    wrapper.find(Button).simulate("submit");
    moxios.wait(() => {
      let request = moxios.requests.mostRecent();
      wrapper.update();
      expect(JSON.parse(request.config.data)).toMatchObject({
        id: 1,
        username: "yash@abc.com",
        password: "yash",
        firstname: "Yash",
        lastname: "IK",
        location: "UK",
        mobileNumber: 12121212,
      });
      done();
    });
  });

  it("Should Have Password Required for Registration", () => {
    expect(wrapper.find(TextField).at(1).props().required).toEqual(true);
  });

  test("Should Render Registration page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
