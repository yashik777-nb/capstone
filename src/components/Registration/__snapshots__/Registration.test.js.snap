// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Component Tests Should Render Registration page 1`] = `ReactWrapper {}`;
